### PR TITLE
Add docs for new SAST policies

### PR DESCRIPTION
**Pull request by tsmithv11**

### CODE REVIEW

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-193.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+----
+const userInput = prompt('Enter your name:');
+const sanitizedInput = DOMPurify.sanitize(userInput);
+document.getElementById('display').innerHTML = sanitizedInput;
+----
+
+In the fixed code, user input is sanitized using `DOMPurify.sanitize()` before being written to the HTML. This ensures that any potentially harmful scripts are removed from the user input, mitigating the risk of XSS attacks.
+
+Additional secure code example with `sanitize-html`:
+
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-193.adoc:61]: This example demonstrates the usage of the `sanitize-html` library to sanitize user input before writing it to the HTML, ensuring that the input is safe from XSS attacks.

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
+== DOM updates should not lead to cross-site scripting (XSS) attacks
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc:1]: Options:

- Unsanitized DOM updates lead to cross-site scripting (XSS) attacks
- DOM updates lead to cross-site scripting (XSS) attacks

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+|https://cwe.mitre.org/data/definitions/79.html[CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')]
+
+|OWASP Categories
+|https://owasp.org/Top10/A03_2021-Injection/[A03:2021 - Injection]
+
+|=== 
+
+=== Description
+
+This policy detects unsafe DOM updates that can lead to Cross-Site Scripting (XSS) attacks in JavaScript and TypeScript applications. XSS vulnerabilities occur when untrusted data is written to the DOM without proper validation or sanitization, allowing attackers to inject malicious scripts.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc:33]: This policy detects whether JavaScript and TypeScript applications perform unsafe DOM updates that can introduce Cross-Site Scripting (XSS) vulnerabilities. XSS vulnerabilities occur when untrusted data is written to the DOM without proper validation or sanitization, allowing attackers to inject malicious scripts.

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+const userInput = document.location.search;
+document.write(userInput);
+document.getElementById('output').innerHTML = userInput;
+----
+
+In this code snippet, user input from `document.location.search` is directly written to the DOM using `document.write` and `innerHTML` without any validation or sanitization. This exposes the application to XSS attacks, where an attacker could inject malicious scripts through the user input.
+
+=== Fix - Buildtime
+
+To fix this issue, ensure that user input is properly sanitized before being written to the DOM. Use libraries such as DOMPurify to sanitize user input and prevent XSS attacks.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-196.adoc]: Ensure that user input is properly sanitized before being written to the DOM. Use libraries such as DOMPurify to sanitize user input and prevent XSS attacks.

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-197.adoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
+== Cryptographic keys should be robust
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-197.adoc:1]: Unrobust cryptographic keys

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-197.adoc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+const { publicKey, privateKey } = crypto.generateKeyPairSync('rsa', {
+  modulusLength: 1024,
+});
+----
+
+In this code snippet, an RSA key pair is generated with a modulus length of 1024 bits, which is considered weak by modern cryptographic standards. Similarly, using insecure named curves for elliptic curve cryptography (ECC) can also lead to vulnerabilities.
+
+=== Fix - Buildtime
+
+To fix this issue, ensure that cryptographic keys are generated with sufficient strength. Use at least 2048-bit keys for RSA and secure named curves for ECC.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-197.adoc:49]: Ensure that cryptographic keys are generated with sufficient strength. Use keys with at least 2048-bit modulus length for RSA. When using ECC, select secure named curves.

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-197.adoc [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+[source,JavaScript]
+----
+const crypto = require('crypto');
+const { publicKey, privateKey } = crypto.generateKeyPairSync('rsa', {
+  modulusLength: 2048,
+});
+----
+
+In the fixed code, the RSA key pair is generated with a modulus length of 2048 bits, providing stronger security.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-197.adoc:61]: Delete-moved to line 52

File: docs/en/enterprise-edition/policy-reference/sast-policies/python-policies/sast-policy-189.adoc [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+app = Flask(__name__)
+
+@app.route('/fetch', methods=['GET'])
+def fetch_data():
+    url = request.args.get('url')
+    response = requests.get(url)
+    return response.content
+----
+
+In this code snippet, user input from `request.args.get('url')` is directly used in an HTTP request via `requests.get(url)` without validation. This exposes the application to SSRF attacks, where an attacker could manipulate the URL to access internal services or sensitive information.
```
Comment (JBakstPaloAlto) [inline docs/en/enterprise-edition/policy-reference/sast-policies/python-policies/sast-policy-189.adoc:54]: Delete-moved to line 39

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-191.adoc [hunk 9]
```diff
@@ -0,0 +1,79 @@
+== The application builds a URL using user-controlled input which can lead to an open redirect vulnerability
```
[the remark this answers is not in the document]
Comment (tsmithv11) [reply docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-191.adoc:1]: Me too. Let's go with the second.

File: docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-193.adoc [hunk 10]
```diff
@@ -0,0 +1,70 @@
+== Input validation is missing for critical functions like innerHTML, outerHTML and document.write can lead to XSS vulnerability
```
[the remark this answers is not in the document]
Comment (tsmithv11) [reply docs/en/enterprise-edition/policy-reference/sast-policies/javascript-policies/sast-policy-193.adoc:1]: I like keeping XSS in the title. Let's go with the second.

**Pull request merged**